models/engine/file_storage.py

```diff
@@ -0,0 +1,61 @@
+"""
+    This module handles serialization/deserialization of objects
+"""
+import json
+from pathlib import Path
+
+class FileStorage:
+    """
+    Serializes instances to a JSON file and deserializes
+    JSON file to instances
+
+    """
+    def __init__(self, file_path, objects):
+        """Instantiates the class with prameters"""
+        self.file_path = file_path
+        self.objects = objects
+
+    @property
+    def file_path(self):
+        """Get/set the file path for the JSON object"""
+        return (self.__file_path)
+
+    @file_path.setter
+    def file_path(self):
+        if not self.__path:
+            self.__path = str(Path.cwd() / objects.json)
+    
+    def objects(self):
+        """Get/set the dictionary that stores all objects by <class name>.id"""
+        return (self.__objects)
+
+    @objects.setter
+    def objects(self):
+        try:
+            if self.__objects:
+                pass
+        except (NameError):
+            self.__objects = {}
+
+    def all(self):
+        """Returns the dictionary __objects"""
+        self.objects()
+
+    def new(self, obj):
+        """sets in __objects the obj with key <obj class name>.id"""
+        class_name = obj['__class__'] 
+        obj_id = obj['id']
+        class_name_id = class_name + '.' + obj_id
+        self.__objects[class_name_id] = obj
+
+    def save(self):
+        """serializes __objects to the JSON file"""
+        obj_file = open(self.__file_path, w)
+        obj_file.write(json.dumps(self.__objects))
+        obj_file.close()
+    
+    def reload(self):
+        """Deserializes the JSON fileto __objects"""
+        if self.__file_path:
+            object_file = open(self.__file)
+            self.__objects = json.load(object_file.read())
```

Comment:
you didn't use "with" here, so I think you should close() the file after line 61?

Reply:
I actually forgot to include that. I would make the changes

Reply:
makes sense. Good job so far

I want us to resolve these PRs and merge to main so I can update main with the latest fully tested (unittests and all ), fully functional BaseModel

I've completed all tests and integrations to ensure we have totally finished tasks 0-4 

- Your next assignment now is to resolve these conversations first so I can merge this to main

- My next assignment now is to pull and merge changes to and from updated main and my branch to get engine file on my branch and update main's base_model module and its tests to my version.

![image](https://user-images.githubusercontent.com/92952014/182997243-51de69e3-14bc-4b84-9651-43f035e32059.png)